backend/src/v1/services/report-calc-service.ts

```diff
@@ -692,6 +692,22 @@ const reportCalcServicePrivate = {
       }
     }
 
+    // If only one gender category has a non-null calculated amount,
+    // reset it to null.
+    if (
+      [
+        meanHourlyPayDiffM,
+        meanHourlyPayDiffF,
+        meanHourlyPayDiffX,
+        meanHourlyPayDiffU,
+      ].filter((d) => d !== null).length < 2
+    ) {
```

Comment:
Can we have a function which has this repeated code and is used where it is needed?

Reply:
Sure.  I'll make it so.